tests/rbd_mirror/test_mirroring_toggle_journaling_feature.py

```diff
@@ -0,0 +1,217 @@
+    if out or err:
+        raise Exception(f"Journaling feature disable failed for {image}")
+
+    image_spec = {"image-or-snap-spec": f"{pool}/{image}"}
+
+    out, err = rbd2.info(**image_spec)
+
+    if out or err:
+        log.info(
+            "Image not present in secondary when journaling is disabled in primary"
+        )
+    else:
+        raise Exception(
+            "Image present in secondary even though journaling is disabled in primary"
+        )
+
+    sleep(15)
+
+    out, err = rbd1.feature.enable(**feature_spec)
+    if out or err:
+        raise Exception(f"Journaling feature enable failed for {image}")
+
+    sleep(15)
+
+    return 0
+
+
+def test_toggle_journaling_feature(**kw):
+    """ """
+    for pool_type in kw.get("pool_types"):
+        config = kw.get("config", {}).get(pool_type)
+        multi_pool_config = getdict(config)
+        for pool, pool_config in multi_pool_config.items():
+            multi_image_config = getdict(pool_config)
+            for image in multi_image_config.keys():
+                try:
+                    with parallel() as p:
+                        p.spawn(
+                            create_delete_snapshot,
+                            rbd=kw.get("rbd_obj"),
+                            pool=pool,
+                            image=image,
+                            operation="create",
+                            count=25,
+                        )
+                        p.spawn(
+                            toggle_journaling_and_verify,
+                            rbd1=kw.get("rbd_obj"),
+                            rbd2=kw.get("sec_obj"),
+                            pool=pool,
+                            image=image,
+                        )
+
+                    rbd2 = kw.get("sec_obj")
+                    wait_for_replay_complete(
+                        rbd=rbd2,
+                        cluster_name=kw.get("ceph_cluster").name,
+                        imagespec=f"{pool}/{image}",
+                    )
+
+                    image_spec = {"image-spec": f"{pool}/{image}", "format": "json"}
+                    out, err = rbd2.snap.ls(**image_spec)
+                    if err:
+                        log.error("Fetch snapshots for secondary faled")
+                        return 1
+                    if out:
+                        out_json = json.loads(out)
+                        snap_json = [k for k in out_json if "snap" in k["name"]]
+                        if len(snap_json) == 25:
+                            log.info(
+                                "All snapshots created at primary are synced to secondary"
+                            )
+                        else:
+                            log.error(
+                                "There is discrepancy between primary and secondary snapshots"
+                            )
+                            return 1
+
+                    with parallel() as p:
```

Comment:
this parallel got repeated why?

Reply:
Once for creating snapshots and verifying and second time for deleting snapshots and verifying. If you observe there is difference in the inputs passed to the parallel methods.

Reply:
Changed it into for loop as discussed.